### ISSUE TITLE
"Remind me of this email" feature

### ISSUE THREAD

**Issue by mityax** [thunderbird/thunderbird-android#7658]
### Checklist

- [X] I have used the search function to see if someone else has already submitted the same feature request.
- [X] I will describe the problem with as much detail as possible.
- [X] This issue only contains a request for one single feature, **not** multiple (related) features.

### App version

6.710

### Problem you are trying to solve

Receiving new emails that imply some task/response one would like to work on later is annoying as
 - one can either open the email but then needs to use a separate todo app to remind oneself about it later or,
 - one could refrain from reading the full email to make the notification stay there in order to not forget working on it later.

### Suggested solution

A "Remind me of this email" feature would solve the issue.
This has been worked on in this really old PR: https://github.com/thunderbird/thunderbird-android/pull/1849 - however it was then closed, by reasoning as follows:

> 1. I think "hiding" messages is an important part of a snooze feature. How that's accomplished technically is another matter (I like the 'move to another folder' approach).
> 2. The 'notify again' time doesn't seem to be persisted anywhere. That'll break the feature e.g. when the device reboots. In my opinion a reminder functionality that doesn't work reliably is worse than none at all.
> 3. Right now nothing is preventing snoozed messages from being removed from the device, e.g. when they fall outside the sync window.
> 4. There's no visual indicator for a snoozed message. Also, no way to un-snooze.

Regarding point 1. and 4., I think it boils down to personal preference/the idea one has of the feature whether "snoozed" emails should be hidden. If it's more a "remind me" feature (like Slack has), than a "snooze this email" feature, I see no need at all to hide the email. In fact, I'd think it's better UX to not hide and just re-trigger the notification after a while as one can then still read the email (e.g. while on the go), ponder about what to do with it, and therefore work more efficiently on it later (when being back home). Therefore, at least as a first step I think a remind me feature could/should be implemented without hiding the email in the app. This is a lot less work too, as I understood it. Consequently, point 4. is also not strictly necessary anymore then - a little icon indicating that a reminder has been set would surely be nice but is not _required_ to obtain a solid UX in my eyes.

Regarding point 3.: This is a valid point and I don't know how much work it is to prevent these messages from being removed from the device or to re-fetch them later. Maybe some internal progress since 2016 has also improved this situation in k9 already (?). In either case, while having/making the email still available when the user clicks the notification would surely be better, I think it wouldn't be a deal breaker for this feature if the email is gone then - this should rarely happen as I'd assume the "remind me" feature is rather intended for short term uses (a few hours) than days, making it unlikely emails disappear that fast. Also - is there any difference to leaving notifications intentionally open for that time in the current app? Is there anything that prevents those emails from being removed from the device? If not then this should not be a requirement for this feature either.

This makes point 2. the only remaining blocker for the feature implementation from the PR. But from my experience, I'd assume scheduling a permanent reminder is not that much of an effort (I have never worked with the k9 codebase though).

I opened this as a new issue to start a discussion on what UX is intended for this feature as suggested in the PR from 2016. There has been [this](https://github.com/thunderbird/thunderbird-android/issues/1549) related issue, but I don't think the discussion there has been very constructive. It has been closed because of the different interaction model connected to a snooze feature - however this does not apply to a "remind me" feature as I described here.

I really think this feature would be extremely useful to lots of people. It would definitely save me from a lot of (if not any) worries and annoyances in my day to day life with k9 mail.

That being said, weighting estimated implementation effort against usefulness, I would definitely not say this feature should be low priority (as the snooze feature was classified to be before).

### Screenshots / Drawings / Technical details

A popup to select reminder time can be opened by either of the following means:
- clicking an action on the push notification of an email
- tapping a context menu action of an email in the inbox list view
- swiping an email to the right in the inbox list view
- tapping a toolbar button or menu entry in the email details page.

In the popup, one can select common reminder times like "30 mins", "1 hour" etc. Optionally, more "sophisticated" times like "this evening, 19pm" or "tomorrow morning, 9am" could also be suggested. Potentionally, these times could be configurable in the app settings. Also potentionally there could be an Option to set a custom reminder time in the popup.

**Comment by mendhak** [comment]
Wanted to support this issue, and am hoping with the new stewardship this is seen in a more favourable light. 

In terms of interaction/UX, in the recent Thunderbird Beta, one of the first things I attempted to do was to left/right swipe an email and see if I could snooze it. Having the ability to snooze as an option in gestures would definitely be useful ("I need to think about this email, OK let me do it tomorrow... <swipe> tomorrow/custom date"). 

Regarding hiding vs not hiding, I don't use Slack so the unhidden-email-reminder isn't a workflow I'm familiar with, actually thinking about it I'm finding it more confusing than not. Myself I would expect the email to be hidden away when I snooze it, simply because that's what Gmail/Outlook do. I don't think I mind where it goes, I have seen various email clients create their own 'snooze' folders and put the email there, presumably to allow me to unsnooze them.

**Comment by mityax** [comment]
I agree that the swipe to trigger the reminder/snooze is a very nice UI, thus I added it to the UI suggestion in the original issue description. 

Regarding the hide vs unhide debate, as I said, I still think this boils down to personal preference – but due to these points, I'd argue that not hiding it is, as far one can objectively talk about this, the better option:
 - one can still read emails to think about what to do with them later, which one cannot do when they're hidden
 - it avoids the extra implementation work for hiding the emails pointed out in the previous PR
 - when adding an indicator that a reminder has been set for the email (and potentially also showing a snackbar to indicate success after setting the reminder), I wouldn't see how this UX could be confusing in any way

So overall, I'd say that hiding the email, while being more consistent with what Gmail and Outlook do, doesn't really add any other value than that – in fact it actually takes away the benefit of being able to access the email at all times. Therefore I'd still advocate for implementing this feature without hiding the email. If hiding the email should prove to be desired by most users, one can still develop this change later, on top of the UI suggested here (maybe even optionally using a toggle in the app settings). This way, there's less blockers for the core issue and an implementation should be doable sooner too.

**Comment by kewisch** [comment]
Hey folks, thanks for filing this issue, it sounds like an interesting idea. Would you mind filing this at https://connect.mozilla.org/t5/ideas/idb-p/ideas/label-name/thunderbird%20android and then closing this issue? Connect provides better opportunities to find product ideas and have users vote on them.

**Comment by mendhak** [comment]
Ok cheers I do see a similar ish idea here, would be good to vote on it then.

https://connect.mozilla.org/t5/ideas/add-the-ability-to-snooze-emails-on-thunderbird-android/idi-p/73169

**Comment by kewisch** [comment]
Thanks for filing! Given we have the request on connect now I am closing here to focus on active development. We'll look at the feature and its votes in our roadmap planning.

**Comment by wsmwk** [comment]
This can be voted at https://connect.mozilla.org/t5/ideas/add-the-ability-to-snooze-emails-on-thunderbird-android/idi-p/73169 - currently ranked #10 of most votes.
